lib/lighting1.js

```diff
@@ -1,57 +1,129 @@
 module.exports = Lighting1;
 /*jshint -W104 */
-const defines = require('./defines');
+const defines = require('./defines'),
+    util = require('util');
 
 /*
  * This is a class for controlling Lighting1 lights.
  */
 function Lighting1(rfxcom, subtype) {
-    var self = this;
+        var self = this;
 
-    self.rfxcom = rfxcom;
-    self.subtype = subtype;
+        self.rfxcom = rfxcom;
+        self.subtype = subtype;
 
-    if (typeof self.subtype === "undefined") {
-        throw new Error("Must provide a subtype.");
+        if (typeof self.subtype === "undefined") {
+            throw new Error("Must provide a subtype.");
+        }
     }
-}
 
 /*
  * Splits the device id into houseCode, unitCode.
  *
- * TODO: Validate houseCode/unitCode.
  */
 Lighting1.prototype._splitDeviceId = function(deviceId) {
-  var parts = deviceId.split(""),
-      houseCode = parts[0].charCodeAt(0),
-      unitCode = parseInt(parts.slice(1).join(""));
-  return {
-      houseCode: houseCode,
-      unitCode: unitCode
-  };
-};
-
-Lighting1.prototype._sendCommand = function(deviceId, command, callback) {
-    var self = this,
-        device = self._splitDeviceId(deviceId),
-        cmdId = self.rfxcom.getCmdNumber(),
-        buffer = [0x07, defines.LIGHTING1, self.subtype, cmdId,
-                  device.houseCode, device.unitCode, command, 0];
-
-    if (self.rfxcom.options.debug) {
-        console.log("Sending %j", self.rfxcom.dumpHex(buffer));
-    }
-    self.rfxcom.serialport.write(buffer, function(err, response) {
-        if (typeof callback === "function") {
-            callback(err, response, cmdId);
+        var parts, houseCode, unitCode;
+        if (util.isArray(deviceId)) {
+            parts = deviceId;
+            if (parts.length !== 2) {
+                throw new Error("Invalid deviceId format");
+            }
+            unitCode = parseInt(parts[1]);
+        } else {
+            parts = deviceId.split("");
+            unitCode = parseInt(parts.slice(1).join(""));
+        }
+        houseCode = parts[0].toUpperCase().charCodeAt(0);
+        if (houseCode < 0x41 || (this.subtype === 0x04 && houseCode > 0x43) ||
+            ((this.subtype === 0x06 || this.subtype === 0x0a) && houseCode > 0x44) || houseCode > 0x50) {
+            throw new Error("Invalid house code '" + parts[0] + "'");
+        }
+        if  (((this.subtype === 0 || this.subtype === 1 || this.subtype === 3) && unitCode > 16) ||
```

Comment:
I've managed to do that by requiring './index.js' and adding a new function to the prototype:

    Lighting1.prototype.isSubtype = function (subtypeName) {
        return index.lighting1[subtypeName] === this.subtype;
    };

which is used like:

        if (this.isSubtype("X10")) {
            return this._sendCommand(deviceId, 0x02, callback);
        } else {
            throw new Error("Device does not support decreaseLevel()")
        }

I've updated all the lighting objects similarly